libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -157,27 +187,79 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
   Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<FeatureFlagValue>[] {
-    const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+  ): SetConfigurationSettingParam<MsFeatureFlagValue>[] {
+    const settings = new Array<SetConfigurationSettingParam<MsFeatureFlagValue>>();
+    const featureFlags = new Array<MsFeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
-        );
+    if (this.dotnetFmSchemaKeyWord) {
+      const dotnetSchemaFeatureFlags = this.getDotnetFmSchemaFeatureFlags(config);
+
+      if (dotnetSchemaFeatureFlags) {
+        const featureManagementIndex = Constants.FeatureManagementKeyWords.indexOf(this.dotnetFmSchemaKeyWord);
+
+        for (const featureFlag in dotnetSchemaFeatureFlags) {
```

Comment:
To confirm, if `dotnetSchemaFeatureFlags` is always an object, do we need the if statement on line 197, since for loop won't be executed on an empty object

Reply:
Updated as discussed offline